src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php

```diff
@@ -39,6 +39,18 @@ public function __construct(EntityManager $entityManager, ClassMetadata $class)
         $this->associationHydrator = new AssociationHydrator($entityManager, $class);
     }
 
+    public function findOrderById(string $id): ?OrderInterface
```

Comment:
```suggestion
    public function findCartOrderById(string $id): ?OrderInterface
```
?

Reply:
I'm in favour of the current name, but lets see what others have to say

Reply:
No hard feelings :).

Although I would argue that with the current naming and what the code actually does I would expect:
```php
public function findOrderById(string $id, string $state = OrderInterface::STATE_CART): ?OrderInterface
```

otherwise it might be confusing/inconsistent.

Reply:
I mean in my opinion the best solution would be to have two repositories. One for orders and one for carts but that is too late now.

But the last suggestion of @HeahDude with the second parameter also sounds good.

Reply:
oh, now I get your suggestion - for me naming was ok because for me order is in a different state than "cart" - I'm open to refactoring PRs

Reply:
> Although I would argue that with the current naming and what the code actually does I would expect:
> 
> ```
> public function findOrderById(string $id, string $state = OrderInterface::STATE_CART): ?OrderInterface
> ```
> 
> otherwise it might be confusing/inconsistent.

It's indeed confusing, even if I'm familiar with `findCartById` construction, but what you're offering is more of `findInStateById` type of method.

As someone who on boards non-sylius dev to sylius projects from time to time, I can guarantee `findByOrder` will not be understood and might lead to methods like `findByTaxon` in taxon repository as made-by-example. :D

One more option: `findPlacedOrderBy***`/`findCartBy***`

Reply:
I would stick to Doctrine's magic method convention and name it `findOneByIdAndState`.
Also, it should be added to the interface no matter its name.